syntax = "proto3";

//option java_multiple_files = true;
//option java_package = "io.grpc.examples.helloworld";
//option java_outer_classname = "HelloWorldProto";
//option objc_class_prefix = "HLW";

package wordgame;

// The greeting service definition.
service Game {
  rpc ChoosePhrase (PhraseRequest) returns (PhraseResultReply) {}
  rpc InitializePhrase (InitRequest) returns (InitResultReply) {}
  rpc ChangeLetter (LetterRequest) returns (LetterResultReply) {}
}


message PhraseRequest {
  string file_name = 1;
}

message PhraseResultReply {
  string phrase = 1;
}

message InitRequest {
  string init_request_phrase = 1;
}

message InitResultReply {
  string init_result_phrase = 1;
}

message LetterRequest {
  string chosen_phrase = 1;
  string initialized_phrase = 2;
  string letter = 3;
}

message LetterResultReply {
  string result = 1;
}
